File: javasrc/elastic_solid/DataWrite.java
import java.io.*; // Gets us CSV output
import java.util.List; // Gets us lists

/*
	Contains:
		class: DataWrite
			(functions on writing data for analysis)
			
			function: MakeHeaders
				(write headers for csvs)
			function: Node
				(Writes out node data to a csv)
			function: MaterialPoint
				(writes out material point data to a csv)
*/


public class DataWrite {
	

	public static void MakeHeaders(
		FileWriter myFile,
		List<String> titles
	) throws IOException {


		for ( int i = 0; i < titles.size(); ++i ) {


			myFile.append( titles.get(i) );

			if ( i != titles.size() - 1 ) {

				myFile.append(',');

			}
			else {

				myFile.append('\n');

			}//end if


		}//end for


	}//end MakeHeaders


	// Writes node data to a file at a given time
	public static void Node(
		FileWriter nodeData,
		List<Node> nodes,
		double t
	) throws IOException{


		
		// Write the time and node quantities
		for ( int i = 0; i < Constants.num_nodes; ++i ) {


			nodeData.append( String.valueOf(t) );
			nodeData.append(',');
			nodeData.append( String.valueOf(nodes.get(i).xpos) );
			nodeData.append( ',' );
			nodeData.append( String.valueOf(nodes.get(i).stress) );
			nodeData.append( ',' );
			nodeData.append( String.valueOf(nodes.get(i).strain) );
			nodeData.append( ',' );
			nodeData.append( String.valueOf(nodes.get(i).dens) );
			nodeData.append( ',' );
			nodeData.append( String.valueOf(nodes.get(i).xvel) );
			nodeData.append( '\n' );


		}//end for

		// Then leave for later

	}//end Node



	// Write material point data at a given time
	public static void MaterialPoint(
		FileWriter mpData,
		List<MaterialPoint> mps,
		double t
	) throws IOException{


		// Write material point data
		for ( int i = 0; i < Constants.num_particles; ++i ) {


			mpData.append( String.valueOf(t) );
			mpData.append(',');
			mpData.append( String.valueOf(mps.get(i).xpos) );
			mpData.append( ',' );
			mpData.append( String.valueOf(mps.get(i).mass) );
			mpData.append( ',' );
			mpData.append( String.valueOf(mps.get(i).stress) );
			mpData.append( ',' );
			mpData.append( String.valueOf(mps.get(i).strain) );
			mpData.append( ',' );
			mpData.append( String.valueOf(mps.get(i).xvel) );
			mpData.append( '\n' );


		}//end for


	}// end MaterialPoint



}//end DataWrite

File: csrc/elastic_solid/data_write.c
#include "data_write.h"



//This function outputs the CSV we use for data analysis
//	Okay, this one is really hardcoded; we should have 2 different
//	functions that write each respective header.
//	But in this case, our titles are all the same length.
void write_header(
	FILE *myFile,
	char* myTitles[]
) {


	for ( int i = 0; i < nd_titles_size; ++i ) {


		fprintf( myFile, "%s", myTitles[i] );

		if ( i != nd_titles_size-1 ) {

			fprintf( myFile, "," );

		}
		else {

			fprintf( myFile, "\n" );

		}//end if


	}//end for


}//end write_header




// Write the node data for a given time
void node_data_write(
	FILE *nodeData,
	struct node nodes[],
	double t
) {


	//Actually writing the data
	//Basically just go through all the class elements of the node and save them
	for (int i = 0; i < num_nodes; ++i) {
		

		fprintf( nodeData, "%f,", t );
		fprintf( nodeData, "%f,", nodes[i].xpos );
		fprintf( nodeData, "%f,", nodes[i].stress );
		fprintf( nodeData, "%f,", nodes[i].strain );
		fprintf( nodeData, "%f,", nodes[i].dens );
		fprintf( nodeData, "%f\n", nodes[i].xvel );
		

	}//end for


}//end node_data_write






// Write the material point data for a given time
void mp_data_write(
	FILE *mpData,
	struct material_point mps[],
	double t
) {


	//Actually writing the data
	//Basically just go through all the class elements of the node and save them
	for (int i = 0; i < num_particles; ++i) {
		
		
		fprintf( mpData, "%f,", t );
		fprintf( mpData, "%f,", mps[i].xpos );
		fprintf( mpData, "%f,", mps[i].mass );
		fprintf( mpData, "%f,", mps[i].stress );
		fprintf( mpData, "%f,", mps[i].strain );
		fprintf( mpData, "%f\n", mps[i].xvel );


	}//end for


}//end node_data_write